tendermint/spn
# feat(`campaign`): add `CampaignCreationFee` param

Closes #538 

### What does this PR does?

Adds a `CampaignCreationFee` param (an `sdk.Coins`) to `campaign`. This fee is deducted at campaign creation if set (which means creation is prevented if account balance is not sufficient). Default value is an empty `sdk.Coins`, so no fee.

Collected fees are used to fund the `CommunityPool` hence why the need for including the `distribution` keeper to `campaign`.

### How to test?

Set a meaningful value for `CampaignCreationFee` param. Try creating a campaign with or without the sufficient balance to cover for the fee set. The latter case should fail.

## Checklist

These are the criteria that every PR should meet, please check them off as you
review them:
